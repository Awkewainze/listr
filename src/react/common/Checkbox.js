import React from 'react';
import './Checkbox.scss';

const Checkbox = ({
  className = '',
  id,
  checked = false,
  onChange
}) => {
  return (
    <div className={`Checkbox ${className}`}>
      <input type="checkbox" className="Checkbox__hidden" id={id} checked={checked} onChange={onChange}/>
      <label htmlFor={id} className={`Checkbox__styled ${checked ? 'Checkbox__checked' : ''}`}>
        <svg className="Checkbox__icon" viewBox="0 0 24 24">
          <polyline points="20 6 9 17 4 12" />
        </svg>
      </label>
    </div>
  );
};

export default Checkbox;