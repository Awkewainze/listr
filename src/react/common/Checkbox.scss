.Checkbox {
  display: inline-block;
  vertical-align: middle;
  width: 16px;
  margin: 0 auto;

  .Checkbox__hidden {
    border: 0;
    clip: rect(0 0 0 0);
    clip-path: inset(50%);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    white-space: nowrap;
    width: 1px;

    &:focus + & {
      box-shadow: 0 0 0 3px pink;
    }
  }

  .Checkbox__styled {
    display: inline-block;
    width: 16px;
    height: 16px;
    background-color: papayawhip;
    border-radius: 3px;
    transition: all 150ms;
    cursor: pointer;

    &.Checkbox__checked {
      background-color: salmon;

      .Checkbox__icon {
        visibility: visible;
      }
    }

    .Checkbox__icon {
      fill: none;
      stroke: white;
      stroke-width: 2px;
      visibility: hidden;
    }
  }
}