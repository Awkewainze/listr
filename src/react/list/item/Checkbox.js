import React from 'react';
import './Checkbox.scss';

const Checkbox = ({
  className,
  checked,
  index,
  onChange
}) => (
  <div className={`Checkbox ${className}`}>
    <input className="Checkbox__toggle" type="checkbox" id={`Checkbox-${index + 1}`} onChange={onChange} checked={checked}/>
    <label className="Checkbox__toggleLabel" htmlFor={`Checkbox-${index + 1}`}/>
  </div>
);

export default Checkbox;