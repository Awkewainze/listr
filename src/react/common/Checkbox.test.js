import { shallow } from 'enzyme';
import Checkbox from './Checkbox';
import React from 'react';

const TEST_PROPS = {
  onChange: jest.fn(),
  className: 'checkbox',
  id: 'foo',
  checked: false
};

it('matches snapshot', () => {
  expect(shallow(<Checkbox {...TEST_PROPS} />)).toMatchSnapshot();
});

it('is unchecked correctly', () => {
  const props = {
    ...TEST_PROPS,
    checked: false
  };
  const container = shallow(<Checkbox {...props} />);
  expect(container.find('input[type="checkbox"]').prop('checked')).toBeFalsy();
});

it('is checked correctly', () => {
  const props = {
    ...TEST_PROPS,
    checked: true
  };
  const container = shallow(<Checkbox {...props} />);
  expect(container.find('input[type="checkbox"]').prop('checked')).toBeTruthy();
});

it('triggers on change correctly', () => {
  const container = shallow(<Checkbox {...TEST_PROPS} />);
  const expectedEvent = new window.Event('change');
  const { onChange } = TEST_PROPS;
  container.find('input[type="checkbox"]').simulate('change', expectedEvent);
  expect(onChange).toHaveBeenCalledTimes(1);
  expect(onChange).toHaveBeenCalledWith(expectedEvent);
});