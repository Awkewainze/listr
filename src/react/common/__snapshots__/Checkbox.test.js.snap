// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Checkbox checkbox"
>
  <input
    checked={false}
    className="Checkbox__hidden"
    id="foo"
    onChange={[MockFunction]}
    type="checkbox"
  />
  <label
    className="Checkbox__styled "
    htmlFor="foo"
  >
    <svg
      className="Checkbox__icon"
      viewBox="0 0 24 24"
    >
      <polyline
        points="20 6 9 17 4 12"
      />
    </svg>
  </label>
</div>
`;
